src/main/java/org/casbin/jcasbin/main/SyncedEnforcer.java

```diff
@@ -180,10 +180,10 @@ public void buildRoleLinks() {
      */
     public boolean enforce(Object... rvals) {
         try {
-            READ_WRITE_LOCK.writeLock().lock();
+            READ_WRITE_LOCK.readLock().lock();
             return super.enforce(rvals);
```

Comment:
I noticed that in `super.enforce(rvals)`, member variable `aviatorEval` is changed, would that cause thread safety issue?

Reply:
It is possible. I have added synchronization blocks where thread safety problems may occur. I wonder if this will solve the problem

Reply:
Maybe you could try this to avoid every thread entering the synchronized block:
`if (aviatorEval == null || modelModCount != model.getModCount()) {
            synchronized (this) {
                if (aviatorEval == null || modelModCount != model.getModCount()) {`

Reply:
I've never seen this before...

Reply:
It's double checked locking.